incubator/hnc/internal/reconcilers/object_test.go

```diff
@@ -45,18 +45,46 @@ var _ = Describe("Secret", func() {
 		cleanupObjects(ctx)
 	})
 
-	// Exceptions have not been implemented, so we are ignoring the following four tests for now.
+	// Exceptions have not been implemented, so we are ignoring some of the following tests for now.
 	// They should be enabled once exception is done.
-	PIt("should propagate object only to selected namespace using treeSelect", func() {
+	It("should propagate object only to selected namespace using treeSelect", func() {
 		setParent(ctx, barName, fooName)
 		setParent(ctx, bazName, fooName)
 
 		// Create a Role that does NOT propogate to bazName
 		a := map[string]string{"propagate.hnc.x-k8s.io/treeSelect": "!" + bazName}
 		makeObjectWithAnnotation(ctx, "Role", fooName, "testrole", a)
 
-		Eventually(hasObject(ctx, "Role", barName, "testrole")).Should(BeTrue())
-		Consistently(hasObject(ctx, "Role", bazName, "testrole")).Should(BeFalse())
+		Eventually(hasObject(ctx, "Role", barName, "testrole")).Should(BeTrue(), "When propagating testrole to %s", barName)
+		Consistently(hasObject(ctx, "Role", bazName, "testrole")).Should(BeFalse(), "When propagating testrole to %s", bazName)
+	})
+
+	It("should propagate object only to selected namespace using multiple treeSelect", func() {
+		setParent(ctx, barName, fooName)
+		setParent(ctx, bazName, fooName)
+
+		// Create a Role that does NOT propogate to bazName
+		a := map[string]string{"propagate.hnc.x-k8s.io/treeSelect": "!" + bazName + ", !" + barName}
+		makeObjectWithAnnotation(ctx, "Role", fooName, "testrole", a)
+
+		Consistently(hasObject(ctx, "Role", barName, "testrole")).Should(BeFalse(), "When propagating testrole to %s", barName)
+		Consistently(hasObject(ctx, "Role", bazName, "testrole")).Should(BeFalse(), "When propagating testrole to %s", bazName)
+	})
+
+	It("should remove object in selected namespace when treeSelector is updated", func() {
+		setParent(ctx, barName, fooName)
+		setParent(ctx, bazName, fooName)
+
+		makeObject(ctx, "Role", fooName, "testrole")
+		Eventually(hasObject(ctx, "Role", barName, "testrole")).Should(BeTrue(), "When propagating testrole to %s", barName)
+		Eventually(hasObject(ctx, "Role", bazName, "testrole")).Should(BeTrue(), "When propagating testrole to %s", bazName)
+
+		// Update the Role so that it does NOT propogate to bazName
+		a := map[string]string{"propagate.hnc.x-k8s.io/treeSelect": "!" + bazName}
+		Expect(updateObjectWithAnnotation(ctx, "Role", fooName, "testrole", a)).Should(Succeed())
+
+		Consistently(hasObject(ctx, "Role", barName, "testrole")).Should(BeTrue(), "When propagating testrole to %s", barName)
+		Eventually(hasObject(ctx, "Role", bazName, "testrole")).Should(BeFalse(), "When propagating testrole to %s", bazName)
```

Comment:
I'd swap the order of these two (here and in similar tests). Line 87 requires the reconciler to actually do something; line 86 is hoping that the reconciler _doesn't_ do something. So waiting for Eventually to pass gives us a bit more confidence that if the reconciler was _going_ to delete the propagated object, it would have already.

Not a huge deal but probably something worth doing (and commenting).